openlibrary/tests/core/test_i18n.py

```diff
@@ -13,24 +15,40 @@ def ungettext(self, message1, message2, n):
         else:
             return self.gettext(message2)
 
+
 class MockLoadTranslations(dict):
     def __call__(self, lang):
         return self.get(lang)
 
     def init(self, lang, translations):
         self[lang] = MockTranslations(translations)
 
+
 class Test_ungettext:
     def setup_monkeypatch(self, monkeypatch):
         self.d = MockLoadTranslations()
         ctx = web.storage()
 
         monkeypatch.setattr(i18n, "load_translations", self.d)
         monkeypatch.setattr(web, "ctx", ctx)
+        self._load_fake_context()
+
+        web.ctx.site = MockSite()
+        monkeypatch.setattr(web, "setcookie", self._setcookie)
+
+    def _load_fake_context(self):
+        app = web.application()
+        env = {"PATH_INFO": "/", "HTTP_METHOD": "GET"}
+        app.load(env)
+
+    def _setcookie(self, name, value):
+        self.cookie = dict(name=name, value=value)
 
     def test_ungettext(self, monkeypatch):
         self.setup_monkeypatch(monkeypatch)
 
+        web.setcookie('i18n_code', 'es')
```

Comment:
I don't see anything that tests if this causes the behavior to change.

Reply:
Sorry, I don't get the point. I added the code above in order for the test of the `ungettext` function to successfully pass. I believe, that the actual behavior of the function remains the same. 

Prior to my commit, the language was supposed to be accessed via the URL (`load_translations(web.ctx.get('lang', 'en'))`) whereas with my changes the language is retrieved from the cookies. Both, actions are done using inherent `web.py` functionality.

Reply:
OK, got it.